### ISSUE TITLE
Update check fails sometimes

### ISSUE THREAD

**Issue by stirante** [Bedrock-OSS/regolith#154]
Once in a while update check stops working and I suspect it might be GitHub rate limit. Maybe we should cache for like a day or so the response or maybe cache when it was checked and skip checking entirely.

Additionally, we should print the error, when `--debug` flag has been specified here:
https://github.com/Bedrock-OSS/regolith/blob/main/main.go#L144

**Comment by stirante** [comment]
Added error printing in https://github.com/Bedrock-OSS/regolith/commit/a6caab30e4f2088aa8f59bc77fd21b615bd02e44

**Comment by stirante** [comment]
I finally got the error and this is the result:
```
[WARN]  Update check failed
[DEBUG] GET https://api.github.com/repos/Bedrock-OSS/regolith/releases/latest: 403 API rate limit exceeded for <ip>. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 28m14s]
```

The good news is that it only happened after updating all dependencies on 0.0.7 (using the old github API method). While testing extensively with the new verion, I wasn't able to exceed rate limit (5k requests per hour).

I added a small commit, that fixes printing pointer instead of an actual error in https://github.com/Bedrock-OSS/regolith/commit/a966725bf2272938a560a8b2e262ce9585f095b8 and I'm marking this as closed.
